test/sslapitest.c

```diff
@@ -3325,18 +3334,20 @@ static int find_session_cb(SSL *ssl, const unsigned char *identity,
     if (find_session_cb_cnt > 2)
         return 0;
 
-    if (serverpsk == NULL)
+    if (serverpsk == NULL || !SSL_SESSION_up_ref(serverpsk))
         return 0;
 
     /* Identity should match that set by the client */
     if (strlen(srvid) != identity_len
             || strncmp(srvid, (const char *)identity, identity_len) != 0) {
         /* No PSK found, continue but without a PSK */
         *sess = NULL;
+
+        SSL_SESSION_free(serverpsk);
+
         return 1;
```

Comment:
This is a normal "success" completion of this function - not an error path. So it seems strange to me to up ref and then free `serverpsk`. It would seem better to me to keep the up_ref where it was originally and just wrap it in an `if`.

Reply:
Sure. Good point.